Expensify/App
# Added attachment download loading indicator

### Details

- Adds a loading indicator for the download activity of Attachments

### Fixed Issues

$ https://github.com/Expensify/App/issues/6253

### Tests
1. Tested downloads for all platforms
2. Tested parallel downloads for the attachments


### QA Steps
1. Go to any chat with attachments (or add a new doc/zip/xlsx, etc.)
2. Click on the download icon
3. You should see an indicator while the file is about to be downloaded
4. Once the download is about to complete indicator should be replaced back with download icon.



### Tested On


### Screenshots


#### Web

<img width="908" alt="web-download-indicator" src="https://user-images.githubusercontent.com/3069065/143524755-1f2478d6-5a7c-4e0a-95bd-c9ddb56fed2e.png">

#### Mobile Web

<img width="515" alt="mweb-download-indicator" src="https://user-images.githubusercontent.com/3069065/143524783-54fb38e9-a45f-414d-aa10-9f3b2fb105c9.png">

#### Desktop

<img width="934" alt="desktop-download-indicator" src="https://user-images.githubusercontent.com/3069065/143524798-0b99fea3-27ed-4d91-8008-9899f76337cb.png">

#### iOS

https://user-images.githubusercontent.com/3069065/143524811-22209eb7-1610-41dd-964e-a4b578e7290a.mov


#### Android

<img width="474" alt="android-download-indicator" src="https://user-images.githubusercontent.com/3069065/143524830-ad2d5101-d4a5-4309-841e-e9006194a240.png">
